### PR TITLE
Add `useFormGroup` upgrade guide

### PR DESCRIPTION
**Pull request by WiXSL**

### CODE REVIEW

File: docs/CreateEdit.md [hunk 1]
```diff
@@ -2330,15 +2330,16 @@ const PostEdit = () => (
 
 ### Grouping Inputs
 
-Sometimes, you may want to group inputs in order to make a form more approachable. You may use a [`<TabbedForm>`](#the-tabbedform-component), an [`<AccordionForm>`](#the-accordionform-component) or you may want to roll your own layout. In this case, you might need to know the state of a group of inputs: whether it's valid or if the user has changed them (dirty/pristine state).
+Sometimes, you may want to group inputs in order to make a form more approachable. You may use a [`<TabbedForm>`](#the-tabbedform-component), an [`<AccordionForm>`](#the-accordionform-component) or you may want to roll your own layout. In this case, you might need to know the state of a group of inputs: whether it's valid or if the user has changed them (dirty/touched state).
```
Comment (WiXSL) [inline docs/CreateEdit.md:2333]: I wanted to remove the pristine reference, but I don't know if the change is misleading

Comment (djhi) [reply docs/CreateEdit.md:2333]: No, `pristine` equivalent is `!isDirty`

Comment (djhi) [reply docs/CreateEdit.md:2333]: Where final-form was returning multiple boolean for the same thing, hook-form only return one

File: docs/CreateEdit.md [hunk 2]
```diff
@@ -2330,15 +2330,16 @@ const PostEdit = () => (
...
 For this, you can use the `<FormGroupContextProvider>`, which accepts a group name. All inputs rendered inside this context will register to it (thanks to the `useInput` hook). You may then call the `useFormGroup` hook to retrieve the status of the group. For example:
 
 ```jsx
-import { Edit, SimpleForm, TextInput, FormGroupContextProvider, useFormGroup } from 'react-admin';
+import { Edit, SimpleForm, TextInput, FormGroupContextProvider, useFormGroup, minLength } from 'react-admin';
 import { Accordion, AccordionDetails, AccordionSummary, Typography } from '@mui/material';
+import ExpandMoreIcon from '@mui/icons-material/ExpandMoreIcon';
 
-const PostEdit = (props) => (
+const PostEdit = props => (
```
Comment (djhi) [inline docs/CreateEdit.md:2342]:
```suggestion
const PostEdit = () => (
```

File: docs/CreateEdit.md [hunk 3]
```diff
@@ -2330,15 +2330,16 @@ const PostEdit = () => (
...
     <Edit {...props}>
```
Comment (djhi) [inline docs/CreateEdit.md:2343]:
```suggestion
    <Edit>
```

**Review by djhi — CHANGES_REQUESTED** [review]

**Review by djhi — APPROVED** [review]

**Pull request merged**